Setup.pm

```diff
@@ -35,21 +35,46 @@ sub update_win32_shell() {
     return;
 }
 
-sub create_shortcut {
-    my $start_menu_path = Win32::GetFolderPath(Win32::CSIDL_STARTMENU());
-    my $start_menu_base = catfile($start_menu_path, 'ActiveState');
+sub desktop_dir_path() {
+    return Win32::GetFolderPath(Win32::CSIDL_DESKTOPDIRECTORY());
+}
 
-    my $target  = $Config{perlpath};
-    my $icon    = q();
-    my $lnkPath = catfile($start_menu_base, 'Perl.lnk');
+sub start_menu_path() {
+    return Win32::GetFolderPath(Win32::CSIDL_STARTMENU());
+}
 
-    mkpath($start_menu_base);
+sub create_internet_shortcut() {
+    BEGIN { Win32::Unicode::InternetShortcut->CoInitialize(); }
```

Comment:
Putting this inside a sub is meaningless. All `BEGIN` and `END` blocks are executed at interpreter startup and shutdown.